src/app/campaign/store/effects/campaign.effects.ts

```diff
@@ -1,277 +1,287 @@
+          ),
+          // retrieve the flights' days from each created flight response
+          mergeMap(
+            ({
+              campaignDoc,
+              deletedFlightDocs,
+              updatedFlightDocs,
+              createdFlightDocs
+            }: {
+              campaignDoc: HalDoc;
+              deletedFlightDocs: { [id: number]: HalDoc };
+              updatedFlightDocs: { [id: number]: HalDoc };
+              createdFlightDocs: { [id: number]: HalDoc };
+            }) => {
+              const flightDaysDocs: Observable<{ [id: number]: HalDoc[] }> = createdFlightDocs
+                ? forkJoin(
+                    Object.values(createdFlightDocs).reduce(
+                      (acc, doc) => ({ ...acc, [doc.id]: this.campaignService.loadFlightDays(doc) }),
+                      {}
+                    )
+                  )
+                : of({});
+              return flightDaysDocs.pipe(
+                map(createdFlightDaysDocs => ({
+                  campaignDoc,
+                  deletedFlightDocs,
+                  updatedFlightDocs,
+                  createdFlightDocs,
+                  createdFlightDaysDocs
+                }))
+              );
+            }
+          ),
+          // retrieve the flights' days from each updated flight response
+          mergeMap(
+            ({
+              campaignDoc,
+              deletedFlightDocs,
+              updatedFlightDocs,
+              createdFlightDocs,
+              createdFlightDaysDocs
+            }: {
+              campaignDoc: HalDoc;
+              deletedFlightDocs: { [id: number]: HalDoc };
+              updatedFlightDocs: { [id: number]: HalDoc };
+              createdFlightDocs: { [id: number]: HalDoc };
+              createdFlightDaysDocs: { [id: number]: HalDoc[] };
+            }) => {
+              const flightDaysDocs: Observable<{ [id: number]: HalDoc[] }> = updatedFlightDocs
+                ? forkJoin(
+                    Object.values(updatedFlightDocs).reduce(
+                      (acc, doc) => ({ ...acc, [doc.id]: this.campaignService.loadFlightDays(doc) }),
+                      {}
+                    )
+                  )
+                : of({});
+              return flightDaysDocs.pipe(
+                map(updatedFlightDaysDocs => ({
+                  campaignDoc,
+                  deletedFlightDocs,
+                  updatedFlightDocs,
+                  updatedFlightDaysDocs,
+                  createdFlightDocs,
+                  createdFlightDaysDocs
+                }))
+              );
+            }
+          ),
+          tap(
+            ({
+              campaignDoc,
+              deletedFlightDocs,
+              createdFlightDocs
+            }: {
+              campaignDoc: HalDoc;
+              deletedFlightDocs: { [id: number]: HalDoc };
+              createdFlightDocs: { [id: number]: HalDoc };
+            }) => {
+              this.toastr.success('Campaign saved');
```

Comment:
This mega-save effect makes a fun diff when the indentation changes slightly.

But just viewing the file... yeah, looks pretty much the same to me!